python/dgl/graphbolt/utils/sample_utils.py

```diff
@@ -1,11 +1,58 @@
 """Utility functions for sampling."""
 
 from collections import defaultdict
-from typing import Dict, Tuple, Union
+from typing import Dict, List, Tuple, Union
 
 import torch
 
 
+def unique_and_compact(
+    nodes: Union[
+        List[torch.Tensor],
+        Dict[str, List[torch.Tensor]],
+    ],
+):
+    """
+    Compact a list of nodes tensor.
+
+    Parameters
+    ----------
+    nodes : List[torch.Tensor] or Dict[str, List[torch.Tensor]]
+        List of nodes for compacting.
+        the unique_and_compact will be done per type
+        - If `nodes` is a list of tensor: It means the graph is homogeneous,
+        all the tensors will do unique_and_compact together.
+        - If `nodes` is a list of dictionary: The keys should be node type and
+        the values should be corresponding nodes, and the IDs inside are
+        heterogeneous ids. And the unique_and_compact will be done per type.
+
+    Returns
+    -------
+    List[torch.Tensor] or Dict[str, List[torch.Tensor]]
+    """
+    is_heterogeneous = isinstance(nodes, dict)
+
+    def unique_and_compact_per_type(nodes):
+        nums = [node.size(0) for node in nodes]
+        nodes = torch.cat(nodes)
+        empty_tensor = nodes.new_empty(0)
+        unique, compacted, _ = torch.ops.graphbolt.unique_and_compact(
+            nodes, empty_tensor, empty_tensor
+        )
+        compacted = compacted.split(nums)
+        return unique, compacted
```

Comment:
The python doc says it returns 1 item, why there are 2 here?

Reply:
Should be a tuple of 2, have fixed the comment.